typescript/sdk/src/deploy/middleware/TokenBridgeRouterDeployer.ts

```diff
@@ -0,0 +1,224 @@
+    configMap: ChainMap<Chain, TokenBridgeConfig>,
+    protected core: HyperlaneCore<Chain>,
+    protected create2salt = 'tokenbridgedeployersalt',
+  ) {
+    super(multiProvider, configMap, tokenBridgeFactories, {});
+  }
+
+  async enrollRemoteRouters(
+    contractsMap: ChainMap<Chain, TokenBridgeContracts>,
+  ): Promise<void> {
+    // Enroll the TokenBridgeRouter with each other
+    await super.enrollRemoteRouters(contractsMap);
+
+    // Enroll the circle adapters with each other
+    const deployedChains = Object.keys(contractsMap);
+    await promiseObjAll(
+      objMap(contractsMap, async (local, contracts) => {
+        const chainConnection = this.multiProvider.getChainConnection(local);
+        // only enroll chains which are deployed
+        const enrollChains = this.multiProvider
+          .remoteChains(local)
+          .filter((c) => deployedChains.includes(c));
+        for (const remote of enrollChains) {
+          const remoteDomain = chainMetadata[remote].id;
+          if (
+            !contracts.circleBridgeAdapter ||
+            !contractsMap[remote].circleBridgeAdapter
+          ) {
+            continue;
+          }
+          const current = await contracts.circleBridgeAdapter.routers(
+            remoteDomain,
+          );
+          const expected = utils.addressToBytes32(
+            contractsMap[remote].circleBridgeAdapter!.address,
+          );
+          if (current !== expected) {
+            await super.runIfOwner(
+              local,
+              contracts.circleBridgeAdapter,
+              async () => {
+                this.logger(
+                  `Enroll ${remote}'s CircleBridgeAdapter on ${local}`,
+                );
+                await chainConnection.handleTx(
+                  contracts.circleBridgeAdapter!.enrollRemoteRouter(
+                    chainMetadata[remote].id,
+                    expected,
+                    chainConnection.overrides,
+                  ),
+                );
+              },
+            );
+          }
+        }
+      }),
+    );
+  }
+
+  // Custom contract deployment logic can go here
+  // If no custom logic is needed, call deployContract for the router
+  async deployContracts(
+    chain: Chain,
+    config: TokenBridgeConfig,
+  ): Promise<TokenBridgeContracts> {
+    const cc = this.multiProvider.getChainConnection(chain);
+    const initCalldata =
+      TokenBridgeRouter__factory.createInterface().encodeFunctionData(
+        'initialize',
+        [config.owner, config.connectionManager, config.interchainGasPaymaster],
+      );
+    const router = await this.deployContract(chain, 'router', [], {
+      create2Salt: this.create2salt + 'TokenBridgeRouter',
+      initCalldata,
+    });
+
+    const bridgeAdapters: Partial<TokenBridgeContracts> = {};
+
+    for (const adapterConfig of config.bridgeAdapterConfigs) {
+      switch (adapterConfig.type) {
+        case BridgeAdapterType.Circle:
```

Comment:
Nit, switches in JS are actually more verbose and create more indentation than just a series of if statements :)